dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/MessageQueueListenerDataManager.kt

```diff
@@ -0,0 +1,129 @@
+                ),
+                exchange = Exchange(ExchangeName.DataQualityAssured, declare = "false"),
+                key = [RoutingKeyNames.data],
+            ),
+        ],
+    )
+    @Transactional
+    fun updateMetaData(
+        @Payload jsonString: String,
+        @Header(MessageHeaderKey.CorrelationId) correlationId: String,
+        @Header(MessageHeaderKey.Type) type: String,
+    ) {
+        messageQueueUtils.validateMessageType(type, MessageType.QaCompleted)
+        val qaCompletedMessage = objectMapper.readValue(jsonString, QaCompletedMessage::class.java)
+        val dataId = qaCompletedMessage.identifier
+        if (dataId.isEmpty()) {
+            throw MessageQueueRejectException("Provided data ID is empty")
+        }
+        messageQueueUtils.rejectMessageOnException {
+            val metaInformation = metaDataManager.getDataMetaInformationByDataId(dataId)
+            metaInformation.qaStatus = qaCompletedMessage.validationResult
+            if (qaCompletedMessage.validationResult == QaStatus.Accepted) {
+                metaDataManager.setActiveDataset(metaInformation)
+            }
+            logger.info(
+                "Received quality assurance: ${qaCompletedMessage.validationResult} for data upload with DataId: " +
+                    "$dataId with Correlation Id: $correlationId",
+            )
+        }
+    }
+
+    /**
+     * Method that listens to the stored queue and removes data entries from the temporary storage once they have been
+     * stored in the persisted database. Further it logs success notification associated containing dataId and
+     * correlationId
+     * @param dataId the ID of the dataset to that was stored
+     * @param correlationId the correlation ID of the current user process
+     * @param type the type of the message
+     */
+    @RabbitListener(
+        bindings = [
+            QueueBinding(
+                value = Queue(
+                    "dataStoredBackendDataManager",
+                    arguments = [
+                        Argument(name = "x-dead-letter-exchange", value = ExchangeName.DeadLetter),
+                        Argument(name = "x-dead-letter-routing-key", value = "deadLetterKey"),
+                        Argument(name = "defaultRequeueRejected", value = "false"),
+                    ],
+                ),
+                exchange = Exchange(ExchangeName.ItemStored, declare = "false"),
+                key = [RoutingKeyNames.data],
+            ),
+        ],
+    )
+    fun removeStoredItemFromTemporaryStore(
+        @Payload dataId: String,
+        @Header(MessageHeaderKey.CorrelationId) correlationId: String,
+        @Header(MessageHeaderKey.Type) type: String,
+    ) {
+        messageQueueUtils.validateMessageType(type, MessageType.DataStored)
+        if (dataId.isEmpty()) {
+            throw MessageQueueRejectException("Provided data ID is empty")
+        }
+        logger.info("Internal Storage sent a message - job done")
```

Comment:
Der log erscheint mir unnötig, wenn man den hierunter schon hat.

Reply:
Zusammengefasst

Reply:
Ich finde den gesamten log komisch. Wieso loggt das backend, dass der internal storage etwas getan hat.

Würde eher loggen "Received message that dataset with dataId X has been successfully stored. Cor..."
Das reicht meines Erachtens und trifft die Wahrheit besser.

Ob der internal storage mit etwas fertig ist, sollte er selbst loggen.   Das ist mir generell in unseren Dataland logs öfter schon aufgefallen, dass die Logs für andere Services sprechen. Lass uns da mal besser darauf achten.

Reply:
Das kann ich gerne anpassen, auch wenn es natürlich an der Stelle klar ist, welcher Service die Message geschickt hat, da man ja aktiv auf den internal Storage lauscht.

Reply:
Genau. Und weil es klar ist, macht der log ja eben keinen Sinn bzw. stiftet keinen Mehrwert, so wie er vorher war.

Reply:
Internal storage Erwähnung ist raus